test/output directory/output directory/CMakeLists.txt

```diff
@@ -2,30 +2,44 @@ cmake_minimum_required(VERSION 3.15)
 project(test_project VERSION 0.1.0)
 include(../../test_header.cmake)
 
-corrosion_import_crate(MANIFEST_PATH Cargo.toml)
+corrosion_import_crate(MANIFEST_PATH proj1/Cargo.toml)
 
 # Note: The output directories defined here must be manually kept in sync with the expected test location.
-set_target_properties(rust_bin
+set_target_properties(rust_bin1
     PROPERTIES
-        RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_bin"
-        PDB_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_bin_pdb"
+        RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_bin_targetprop"
+        PDB_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_bin_pdb_targetprop"
 
 )
-set_target_properties(rust_lib PROPERTIES ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_archive")
-set_target_properties(rust_lib
+set_target_properties(rust_lib1 PROPERTIES ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_archive_targetprop")
+set_target_properties(rust_lib1
     PROPERTIES
-        LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_lib"
-        PDB_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_lib_pdb"
+        LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_lib_targetprop"
+        PDB_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_lib_pdb_targetprop"
 )
 
-add_custom_command(TARGET cargo-build_rust_bin POST_BUILD
+add_custom_command(TARGET cargo-build_rust_bin1 POST_BUILD
     COMMAND
     ${CMAKE_COMMAND} -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/another_dir"
     COMMAND
-    ${CMAKE_COMMAND} -E copy_if_different "$<TARGET_PROPERTY:rust_bin,LOCATION>" "${CMAKE_CURRENT_BINARY_DIR}/another_dir/moved_bin"
+    ${CMAKE_COMMAND} -E copy_if_different "$<TARGET_PROPERTY:rust_bin1,LOCATION>" "${CMAKE_CURRENT_BINARY_DIR}/another_dir/moved_bin"
     )
 
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_bin_var")
+set(CMAKE_PDB_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_bin_pdb_var")
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_archive_var")
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_lib_var")
+set(CMAKE_PDB_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/custom_lib_pdb_var")
```

Comment:
Currently you are overwriting `CMAKE_PDB_OUTPUT_DIRECTORY` here which is probably not intended. Where you perhaps thinking of `CMAKE_COMPILE_PDB_OUTPUT_DIRECTORY`? I previously ignored the existence of this property, because I was under the impression that Rust does not generate a pdb file for static libraries, but I don't think I double checked that. If rust does generate a pdb file for static libs, then `COMPILE_PDB_OUTPUT_DIRECTORY` is probably the relevant property for that artifact, but I'm honestly not too familiar with windows and implibs.

Reply:
The use of `CMAKE_PDB_OUTPUT_DIRECTORY` was intentional here, as that would otherwise be used to initialized the `PDB_OUTPUT_DIRECTORY` target property that the test sets further above when not using variables.

I think changing from `PDB_OUTPUT_DIRECTORY` to `COMPILE_PDB_OUTPUT_DIRECTORY` could be done in a separate (unrelated) change. For now I think it makes more sense to keep the symmetry.

Reply:
I don't quite follow. From my understanding the assignment in line 32 would overwrite the previous assignment in line 29, so only one of those should be correct.

Reply:
Ahh right. I read too fast :(. I think it's actually slightly more complicated. Let's ignore the static library case for a moment:

The target property test places the pdb for the bin into one directory, and the pdb for the (shared) library into another. That is what the target property allows doing. But with the variable based approach both have to end up in the same expected directory, because the same variable feeds them.